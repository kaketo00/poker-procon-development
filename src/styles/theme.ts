import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
