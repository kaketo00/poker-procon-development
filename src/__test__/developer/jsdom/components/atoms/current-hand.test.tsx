import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import type { CurrentHandProps } from '@/components/atoms/current-hand';
import CurrentHand from '@/components/atoms/current-hand';
import theme from '@/styles/theme';

const Component = (props: CurrentHandProps) => (
  <ChakraProvider theme={theme}>
    <CurrentHand {...props} />
  </ChakraProvider>
);

describe('CurrentHand', () => {
  test('スナップショット/交換あり', () => {
    const { asFragment } = render(
      <Component
        phase="draw-1"
        beforeCards={[
          { suit: 'Clubs', number: 5 },
          { suit: 'Clubs', number: 6 },
          { suit: 'Spades', number: 9 },
          { suit: 'Diamonds', number: 9 },
          { suit: 'Spades', number: 1 },
        ]}
        afterCards={[
          { suit: 'Clubs', number: 5 },
          { suit: 'Clubs', number: 6 },
          { suit: 'Diamonds', number: 9 },
          { suit: 'Spades', number: 9 },
          { suit: 'Hearts', number: 9 },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/交換なし', () => {
    const { asFragment } = render(
      <Component
        phase="draw-1"
        afterCards={[
          { suit: 'Clubs', number: 5 },
          { suit: 'Clubs', number: 6 },
          { suit: 'Diamonds', number: 9 },
          { suit: 'Spades', number: 9 },
          { suit: 'Hearts', number: 9 },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/カード未所持', () => {
    const { asFragment } = render(<Component phase="draw-1" afterCards={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/終了フェーズ', () => {
    const { asFragment } = render(
      <Component
        phase="finished"
        afterCards={[
          { suit: 'Clubs', number: 5 },
          { suit: 'Clubs', number: 6 },
          { suit: 'Diamonds', number: 9 },
          { suit: 'Spades', number: 9 },
          { suit: 'Hearts', number: 9 },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('交換を行った時は前後の役が表示される', () => {
    const { getByText } = render(
      <Component
        phase="draw-1"
        beforeCards={[
          { suit: 'Clubs', number: 5 },
          { suit: 'Clubs', number: 6 },
          { suit: 'Spades', number: 9 },
          { suit: 'Diamonds', number: 9 },
          { suit: 'Spades', number: 1 },
        ]}
        afterCards={[
          { suit: 'Clubs', number: 5 },
          { suit: 'Clubs', number: 6 },
          { suit: 'Diamonds', number: 9 },
          { suit: 'Spades', number: 9 },
          { suit: 'Hearts', number: 9 },
        ]}
      />
    );
    expect(getByText('One Pair → Three of a Kind')).toBeInTheDocument();
  });
});
