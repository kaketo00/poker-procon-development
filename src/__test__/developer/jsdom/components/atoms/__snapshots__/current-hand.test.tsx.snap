// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentHand スナップショット/カード未所持 1`] = `
<DocumentFragment>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`CurrentHand スナップショット/交換あり 1`] = `
<DocumentFragment>
  <div
    class="css-z165it"
  >
    <p
      class="chakra-text css-1876eo0"
    >
      One Pair → Three of a Kind
    </p>
    <img
      alt="bubble"
      class="chakra-image css-13pr30p"
      src="/bubble.svg"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`CurrentHand スナップショット/交換なし 1`] = `
<DocumentFragment>
  <div
    class="css-z165it"
  >
    <p
      class="chakra-text css-1876eo0"
    >
      Three of a Kind
    </p>
    <img
      alt="bubble"
      class="chakra-image css-13pr30p"
      src="/bubble.svg"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`CurrentHand スナップショット/終了フェーズ 1`] = `
<DocumentFragment>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
